### PR TITLE
fix(badge): simplify children type

### PR DESCRIPTION
**Pull request by shleewhite**
- [x] changes the type of `children` prop to `NonNullable<React.ReactNode>` 

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

**Review by zahnster — APPROVED** [review]
Looks good 👍🏼

File: .changeset/tidy-feet-wash.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@twilio-paste/badge': patch
+'@twilio-paste/core': patch
+---
+
+[Badge] Fixed the type of the children prop
```
Comment (shleewhite) [inline .changeset/tidy-feet-wash.md:6]: ❓ Should this be a patch or a minor version? I made it a patch for now because the type is more inclusive now.

Comment (SiTaggart) [reply .changeset/tidy-feet-wash.md:6]: Patch is appropriate. You're fixing a thing. If you were adding noteworthy new features, it'd be a minor

**Review by TheSisb — APPROVED** [review]

File: packages/paste-core/components/badge/src/hooks.tsx [hunk 2]
```diff
@@ -1,11 +1,9 @@
 import * as React from 'react';
 import {useUIDSeed} from '@twilio-paste/uid-library';
 
-import type {BadgeChildren} from './types';
-
 const DEFAULT_ICON_SIZE = 'sizeIcon10';
 
-export const useResizeChildIcons = (children: BadgeChildren): BadgeChildren => {
+export const useResizeChildIcons = (children: React.ReactNode): React.ReactNode => {
```
Comment (andipants12) [inline packages/paste-core/components/badge/src/hooks.tsx:6]: Should this potentially be [`NonNullable<React.ReactNode>`](https://www.typescriptlang.org/docs/handbook/utility-types.html#nonnullabletype)?

**Review by andipants12 — COMMENTED** [review]
One potential change for the resize hook, otherwise lgtm!

File: packages/paste-core/components/badge/src/hooks.tsx (hunk 2 above)
Comment (shleewhite) [reply packages/paste-core/components/badge/src/hooks.tsx:6]: totally! will update.

**Review by andipants12 — APPROVED** [review]

**Pull request merged**